### ISSUE TITLE
[TomoX] BasePrice should depend on tokenDecimal

### ISSUE THREAD

**Issue by thanhnguyennguyen** [BuildOnViction/tomochain-v1#633]
BasePrice should depend on tokenDecimal of baseToken instead of hard code 10^18
common/constants.go:34
```go
var BasePrice = big.NewInt(1000000000000000000) // 1
```

**Comment by thanhnguyennguyen** [comment]
```golang
quoteQuantity = price * amount / 10^(baseDecimal)
```

price in smallest unit of quoteToken
amount in smallest unit of baseToken

**Comment by thanhnguyennguyen** [comment]
refer tomochain/tomox-sdk-ui/issues/107

**Comment by thanhnguyennguyen** [comment]
fixed in #646
